modules/infinite-scroll/infinity.php

```diff
@@ -1329,40 +1329,45 @@ function query() {
 
 			$results['type'] = 'success';
 
-			/**
-			 * Gather renderer callbacks. These will be called in order and allow multiple callbacks to be queued. Once content is found, no futher callbacks will run.
-			 *
-			 * @module infinite-scroll
-			 *
-			 * @since 6.0.0
-			 */
-			$callbacks = apply_filters( 'infinite_scroll_render_callbacks', array(
-				self::get_settings()->render, // This is the setting callback e.g. from add theme support.
-			) );
-
-			// Append fallback callback. That rhymes.
-			$callbacks[] = array( $this, 'render' );
-
-			foreach ( $callbacks as $callback ) {
-				if ( false !== $callback && is_callable( $callback ) ) {
-					rewind_posts();
-					ob_start();
-					add_action( 'infinite_scroll_render', $callback );
-
-					/**
-					 * Fires when rendering Infinite Scroll posts.
-					 *
-					 * @module infinite-scroll
-					 *
-					 * @since 2.0.0
-					 */
-					do_action( 'infinite_scroll_render' );
-
-					$results['html'] = ob_get_clean();
-					remove_action( 'infinite_scroll_render', $callback );
-				}
-				if ( ! empty( $results['html'] ) ) {
-					break;
+			do_action( 'infinite_scroll_render' );
```

Comment:
The action is now triggered twice. Did you mean to add this one here?

Reply:
This is how it used to work, and the bug is created by removing this call. There might be a different way to fix the issue by changing how p2 itself works...